src/main/java/edu/ucsb/cs156/happiercows/controllers/ProfitsController.java

```diff
@@ -0,0 +1,209 @@
+        return profits;
+    }
+
+    @ApiOperation(value = "Get all profits belonging to a user commons as a user")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all/commons")
+    public Iterable<Profit> allProfitsByUserCommonsId(
+            @ApiParam("userCommonsId") @RequestParam Long userCommonsId) {
+        Long userId = getCurrentUser().getUser().getId();
+
+        UserCommons userCommons = userCommonsRepository.findById(userCommonsId).orElseThrow(() -> new EntityNotFoundException(UserCommons.class, userCommonsId));
+
+        if (userId != userCommons.getUserId())
+            throw new EntityNotFoundException(UserCommons.class, userCommonsId);
+
+        Iterable<Profit> profits = profitRepository.findAllByUserCommonsId(userCommonsId);
+
+        return profits;
+    }
+
+    @ApiOperation(value = "Get all profits belonging to a user commons as an admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @GetMapping("/admin/all/commons")
+    public Iterable<Profit> allProfitsByUserCommonsId_admin(
+            @ApiParam("userCommonsId") @RequestParam Long userCommonsId) {
+        Iterable<Profit> profits = profitRepository.findAllByUserCommonsId(userCommonsId);
+        return profits;
+    }
+
+    @ApiOperation(value = "Create a new Profit as admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/admin/post")
+    public Profit postProfit_admin(
+            @ApiParam("profit") @RequestParam long profit,
+            @ApiParam("timestamp") @RequestParam long timestamp,
+            @ApiParam("userCommonsId") @RequestParam long userCommonsId) {
+        UserCommons userCommons = userCommonsRepository.findById(userCommonsId).orElseThrow(() -> new EntityNotFoundException(UserCommons.class, userCommonsId));
+
+        Profit createdProfit = new Profit();
+        createdProfit.setProfit(profit);
+        createdProfit.setUserCommons(userCommons);
+        createdProfit.setTimestamp(timestamp);
+        Profit savedProfit = profitRepository.save(createdProfit);
+        return savedProfit;
+    }
+
+    @ApiOperation(value = "Create a new Profit as user")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+    public Profit postProfit(
+            @ApiParam("profit") @RequestParam long profit,
+            @ApiParam("timestamp") @RequestParam long timestamp,
+            @ApiParam("userCommonsId") @RequestParam long userCommonsId) {
+
+        Long userId = getCurrentUser().getUser().getId();
+        UserCommons userCommons = userCommonsRepository.findById(userCommonsId).orElseThrow(() -> new EntityNotFoundException(UserCommons.class, userCommonsId));
+
+        Profit createdProfit = new Profit();
+        createdProfit.setProfit(profit);
+        createdProfit.setUserCommons(userCommons);
+        createdProfit.setTimestamp(timestamp);
+        Profit savedProfit = profitRepository.save(createdProfit);
+        return savedProfit;
+    }

```

Comment:
What is the user case for a user to create a profit?

Doesn't that just give users the ability to cheat and give themselves money any time they want?